src/test/java/hudson/plugins/xvnc/XvncTest.java

```diff
@@ -0,0 +1,161 @@
+import hudson.model.FreeStyleProject;
+import hudson.model.Hudson;
+import hudson.plugins.xvnc.Xvnc.DescriptorImpl;
+import hudson.slaves.DumbSlave;
+import hudson.tasks.Builder;
+import hudson.util.OneShotEvent;
+
+import java.io.IOException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Bug;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class XvncTest {
+
+    @Rule public JenkinsRule j = new JenkinsRule();
+
+    @Test @Bug(24773)
+    public void distinctDisplaySpaceForSlaves() throws Exception {
+        DumbSlave slaveA = j.createOnlineSlave();
+        DumbSlave slaveB = j.createOnlineSlave();
+
+        FreeStyleProject jobA = j.jenkins.createProject(FreeStyleProject.class, "jobA");
+        jobA.setAssignedNode(slaveA);
+        FreeStyleProject jobB = j.jenkins.createProject(FreeStyleProject.class, "jobB");
+        jobB.setAssignedNode(slaveB);
+
+        fakeXvncRun(jobA);
+        fakeXvncRun(jobB);
+
+        jobA.getBuildersList().add(new Blocker());
+        jobA.scheduleBuild2(0);
+        Blocker.RUNNING.block(1000);
+
+        FreeStyleBuild build = jobB.scheduleBuild2(0).get();
+        j.assertBuildStatusSuccess(build);
+
+        Blocker.DONE.signal();
+    }
+
+    @Test // The number should not be allocated as builds are executed sequentially
+    public void reuseDisplayNumberOnSameSlave() throws Exception {
+        FreeStyleProject p = j.jenkins.createProject(FreeStyleProject.class, "project");
+
+        runXvnc(p).cleanUp = true;;
+
+        j.buildAndAssertSuccess(p);
+        j.buildAndAssertSuccess(p);
+    }
+
+    @Test
+    public void displayBlacklistedOnOneMachineShouldNobBeBlackistedOnAnother() throws Exception {
+        DumbSlave slaveA = j.createOnlineSlave();
+        DumbSlave slaveB = j.createOnlineSlave();
+
+        FreeStyleProject jobA = j.jenkins.createProject(FreeStyleProject.class, "jobA");
+        jobA.setAssignedNode(slaveA);
+        FreeStyleProject jobB = j.jenkins.createProject(FreeStyleProject.class, "jobB");
+        jobB.setAssignedNode(slaveB);
+
+        // blacklist :42 on slaveA
+        runXvnc(jobA).xvnc = "vncserver :$DISPLAY_NUMBER with wrong args so it wont start";
+        j.assertBuildStatus(Result.FAILURE, jobA.scheduleBuild2(0).get());
+
+        // use :42 on slaveB
+        runXvnc(jobB).cleanUp = true;
+        j.buildAndAssertSuccess(jobB);
+    }
+
+    @Test
+    public void blacklistedDisplayShouldStayBlacklistedBetweenBuilds() throws Exception {
+        FreeStyleProject p = j.jenkins.createProject(FreeStyleProject.class, "project");
+
+        // Blacklist
+        runXvnc(p).xvnc = "vncserver :$DISPLAY_NUMBER with wrong args so it wont start";
+        j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
+
+        // Should still fail
+        runXvnc(p).xvnc = "vncserver :$DISPLAY_NUMBER with wrong args so it wont start";
+        j.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0).get());
```

Comment:
Good point. Second command should be valid but still fail as the only allowed display is expected to be blacklisted.